src/Reader/Reader.php

```diff
@@ -207,26 +212,27 @@ public static function import($uri, $etag = null, $lastModified = null)
                     $lastModified = $cache->getItem($cacheId . '_lastmodified');
                 }
                 if ($etag) {
-                    $headers->addHeaderLine('If-None-Match', $etag);
+                    $headers['If-None-Match'] = [$etag];
                 }
                 if ($lastModified) {
-                    $headers->addHeaderLine('If-Modified-Since', $lastModified);
+                    $headers['If-Modified-Since'] = [$lastModified];
                 }
             }
-            $response = $client->send();
+            $response = $client->get($uri, $headers);
```

Comment:
You are right. I was applying LSP rules but the previous behavior was only for Zend client.

I would add a comment in the code describing the previous instanceof operation is an optimization.